sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -67,7 +73,13 @@ object CollateExpressionBuilder extends ExpressionBuilder {
  */
 case class Collate(child: Expression, collationName: String)
   extends UnaryExpression with ExpectsInputTypes {
-  private val collationId = CollationFactory.collationNameToId(collationName)
+  private val collationId = if (!SQLConf.get.collationEnabled) {
+    // This is here if someone enters a query which only has invalid collation name
+    throw QueryCompilationErrors.collationNotEnabledError()
```

Comment:
how can we hit this branch?

Reply:
Query "SELECT 'aaa' COLLATE 'UNKNOWN_COLLATION_NAME'"

Reply:
Comment below is related to this problem and I am not sure how to proceed on it.

Reply:
shouldn't we fail very early in the parser side if collation is not enabled?

Reply:
Problem is when we use collation function in the format of the query above, the check for collationName id is preformed before creating Collate expression.

Reply:
Concretely the call of `AstBuilder.visitCollate` happens

Reply:
I managed to fix this by setting `collationName` to lazy val in visitCollate.

Reply:
Problem still persists for query of the type ```CREATE TABLE t(col STRING COLLATE 'UNKNOWN_COLLATION_NAME') USING parquet```